### ISSUE TITLE
Sliders: Keyboard interactions for Home and End are ambiguous

### ISSUE THREAD

**Issue by 0ddfell0w** [w3c/aria-practices#600]
> Home: Set the slider to the first allowed value in its range.
> End: Set the slider to the last allowed value in its range.
- https://www.w3.org/TR/wai-aria-practices-1.1/#slider

Notice that all the other keyboard interactions describe increasing or decreasing the value. Home and End instead refer to the "first" and "last" allowed value. It would be much clearer to refer to these value as the maximum and minimum values, or in some other way make it clear: should Home should change the slider in the same direction as ArrowDown, or in the same direction as ArrowUp?

**Comment by mcking65** [comment]
Yes, left, up, and home all in one direction and right, down, and end in the other. This should be easy to clear up.

**Comment by jnurthen** [comment]
We should remove reference to the value in the text while we are changing it. The keys do not move the "value" in a certain direction - they move the thumb in a certain direction. In a horizontal slider these normally correspond but in a vertical slider often the highest value is at the bottom and the lowest value at the top which sometimes makes these keystrokes backwards if we are talking about the value.

**Comment by liamcmitchell-sc** [comment]
I came across this today while looking at the slider example: https://www.w3.org/TR/wai-aria-practices/examples/slider/slider-1.html

What was immediately uncomfortable for me as a keyboard user was that up/page up/home and down/page down/end weren't aligned. Instead, home and end swap direction. I don't think I've ever seen that before.

**Comment by carmacleod** [comment]
Note that the [Slider Example](https://w3c.github.io/aria-practices/examples/slider/slider-1.html) keys operate the same as for HTML `<input type="range">`.
To test this, there's a simple range input labelled "Range" about half-way down this page of [HTML element examples](http://html5accessibility.com/html5elements/).

**Comment by liamcmitchell-sc** [comment]
Thanks, should have tried that. Do you know if these best practices intentionally follow browser behavior?

Behavior of horizontal range (in LTR) seems to be consistent across Chrome, FF, Safari, Edge and match the recommendation here. Switch to RTL and all browsers switch direction of min/max visually. All browsers except Safari also switch left/right keys to match visual direction.

Behavior is not consistent for vertical sliders. Chrome/Safari/Edge use left/up/page up/home for increment and right/down/page down/end for decrement. This is different to the ARIA docs and Firefox: right/up/page up/end for increment and left/down/page down/home for decrement (same as horizontal).

[Origin of Firefox behavior.](https://bugzilla.mozilla.org/show_bug.cgi?id=843725) [Chrome (and Safari/Edge?) behavior comes from Windows 7 sliders.](https://bugs.webkit.org/show_bug.cgi?id=52262) Not sure how widely accepted any of these combinations are, or if this is the right place to try to redefine them... but I'll keep going.

What I think confuses the issue is two different axes - numerical and spatial. I think there is a convention to equate numerical increment with spatial up. We see this in type="number" input where the up key will increment. A slider is often numeric but is rendered spatially.

The keys we are talking about have established spatial direction. [Home](https://en.wikipedia.org/wiki/Home_key) is commonly used to navigate to the beginning of a line or document, this is left/up in top-to-bottom, left-to-right scripts. [End](https://en.wikipedia.org/wiki/End_key) is the opposite, navigates to the end of a line or document. [Page up/down](https://en.wikipedia.org/wiki/Page_Up_and_Page_Down_keys) have the same spatial direction ([MS Excel uses page down + modifiers to move to the right](https://support.microsoft.com/en-us/office/keyboard-shortcuts-in-excel-1798d9d5-842a-42b8-9c99-9b7213f0040f)). Apple now uses directional arrows instead of text for these keys ([british](https://en.wikipedia.org/wiki/Apple_Keyboard#/media/File:Apple_Keyboard_with_Numeric_Keyboard_9612.jpg), [chinese](https://en.wikipedia.org/wiki/Apple_Keyboard#/media/File:Apple_Magic_Keyboard_with_Numeric_Keypad_Traditional_Chinese_(Zhuyin_&_Cangjie).jpg)).

If we are using a slider to interact with visually (spatially), I think it is more intuitive to use the established key directions. I see this is not ideal for non-sighted users who would need some hint as to the axis and direction or the control. An alternative is to completely ignore the spatial directions and have keys always align to numeric directions, completely ignoring slider orientation and text direction.

Either way, I want my home and page up keys to move in the same direction 😄
